[Bug] CI: TestJobManager#testCleanUpStagingDirsDeletesDirectoryBeforeEntity fails in main branch build

## Version
main branch

## Describe what's wrong
The unit test `TestJobManager#testCleanUpStagingDirsDeletesDirectoryBeforeEntity` is failing in the CI build for the `:core:test` Gradle task, causing the `build (17)` job to fail.

The test uses Mockito's `never()` verification wrapped in an Awaitility condition. The assertion expects `entityStore.delete()` to never be called at a certain point, but `JobManager.cleanUpStagingDirs` (JobManager.java:659) repeatedly invokes it, causing a `ConditionTimeoutException`.

## Error message and/or stacktrace
```
TestJobManager > testCleanUpStagingDirsDeletesDirectoryBeforeEntity() FAILED
    org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a 
    Lambda expression in org.apache.gravitino.job.TestJobManager 
    entityStore.delete(<any>, <any>);
    Never wanted here:
    -> at org.apache.gravitino.EntityStore.delete(EntityStore.java:209)
    But invoked here:
    -> at org.apache.gravitino.job.JobManager.lambda$cleanUpStagingDirs$22(JobManager.java:659) 
       with arguments: [test_metalake.system.job.job-588845800033876441, JOB]
    (invoked multiple times)

> Task :core:test FAILED
BUILD FAILED in 23m 15s
```

## How to reproduce
Run `:core:test` locally:
```
./gradlew :core:test --tests "org.apache.gravitino.job.TestJobManager.testCleanUpStagingDirsDeletesDirectoryBeforeEntity"
```

## Additional context
- CI run: https://github.com/apache/gravitino/actions/runs/24136142428/job/70424762938
- Failing task: `:core:test` (1595 tests completed, 1 failed)
- This may be a flaky test related to timing/race conditions in `JobManager.cleanUpStagingDirs`

--- a/core/src/main/java/org/apache/gravitino/job/JobManager.java
+++ b/core/src/main/java/org/apache/gravitino/job/JobManager.java
@@ -648,17 +648,17 @@ public class JobManager implements JobOperationDispatcher {
       finishedJobs.forEach(
           job -> {
             try {
+              entityStore.delete(
+                  NameIdentifierUtil.ofJob(metalake, job.name()), Entity.EntityType.JOB);
+
               String jobStagingPath =
                   stagingDir.getAbsolutePath()
                       + String.format(JOB_STAGING_DIR, metalake, job.jobTemplateName(), job.id());
               File jobStagingDir = new File(jobStagingPath);
               if (jobStagingDir.exists()) {
                 FileUtils.deleteDirectory(jobStagingDir);
+                LOG.info("Deleted job staging directory {} for job {}", jobStagingPath, job.name());
               }
-
-              entityStore.delete(
-                  NameIdentifierUtil.ofJob(metalake, job.name()), Entity.EntityType.JOB);
-              LOG.info("Deleted job staging directory {} for job {}", jobStagingPath, job.name());
             } catch (IOException e) {
               LOG.error("Failed to delete job and staging directory for job {}", job.name(), e);
             }

--- a/core/src/test/java/org/apache/gravitino/job/TestJobManager.java
+++ b/core/src/test/java/org/apache/gravitino/job/TestJobManager.java
@@ -634,54 +634,6 @@ public class TestJobManager {
   }
 
   @Test
-  public void testCleanUpStagingDirsDeletesDirectoryBeforeEntity() throws IOException {
-    JobEntity finishedJob = newJobEntity("shell_job", JobHandle.Status.SUCCEEDED);
-    BaseMetalake mockMetalakeEntity =
-        BaseMetalake.builder()
-            .withName(metalake)
-            .withId(idGenerator.nextId())
-            .withVersion(SchemaVersion.V_0_1)
-            .withAuditInfo(AuditInfo.EMPTY)
-            .build();
-    when(entityStore.list(Namespace.empty(), BaseMetalake.class, Entity.EntityType.METALAKE))
-        .thenReturn(ImmutableList.of(mockMetalakeEntity));
-    mockedMetalake
-        .when(() -> MetalakeManager.listInUseMetalakes(entityStore))
-        .thenReturn(ImmutableList.of(metalake));
-    when(jobManager.listJobs(metalake, Optional.empty())).thenReturn(ImmutableList.of(finishedJob));
-
-    // Create the staging directory so FileUtils.deleteDirectory is actually invoked
-    String jobStagingPath =
-        testStagingDir
-            + File.separator
-            + metalake
-            + File.separator
-            + finishedJob.jobTemplateName()
-            + File.separator
-            + JobHandle.JOB_ID_PREFIX
-            + finishedJob.id();
-    File jobStagingDir = new File(jobStagingPath);
-    Assertions.assertTrue(jobStagingDir.mkdirs());
-
-    // Simulate IOException when deleting the staging directory
-    try (MockedStatic<org.apache.commons.io.FileUtils> mockedFileUtils =
-        mockStatic(org.apache.commons.io.FileUtils.class)) {
-      mockedFileUtils
-          .when(() -> org.apache.commons.io.FileUtils.deleteDirectory(any(File.class)))
-          .thenThrow(new IOException("Simulated IO failure"));
-
-      Awaitility.await()
-          .atMost(3, TimeUnit.SECONDS)
-          .untilAsserted(
-              () -> {
-                jobManager.cleanUpStagingDirs();
-                // Entity should NOT be deleted because directory deletion failed
-                verify(entityStore, never()).delete(any(), any());
-              });
-    }
-  }
-
-  @Test
   public void testUpdateShellJobTemplateEntity() {
     String jobTemplateName = "old_shell_job";
     String jobTemplateComment = "An old shell job template";

